lib/src/idx/trees/vector.rs

```diff
@@ -164,62 +149,79 @@ impl Vector {
 	pub(super) fn euclidean_distance(&self, other: &Self) -> Result<f64, Error> {
 		Self::check_same_dimension("vector::distance::euclidean", self, other)?;
 		match (self, other) {
-			(Vector::F64(a), Vector::F64(b)) => Ok(euclidean(a, b)),
-			(Vector::F32(a), Vector::F32(b)) => Ok(euclidean(a, b)),
-			(Vector::I64(a), Vector::I64(b)) => Ok(euclidean(a, b)),
-			(Vector::I32(a), Vector::I32(b)) => Ok(euclidean(a, b)),
-			(Vector::I16(a), Vector::I16(b)) => Ok(euclidean(a, b)),
-			(Vector::I8(a), Vector::I8(b)) => Ok(euclidean(a, b)),
-			_ => Err(Error::Unreachable),
-		}
-	}
-
-	pub(super) fn cosine_similarity(&self, other: &Self) -> Result<f64, Error> {
```

Comment:
The distances function from the `Distances` crate works for some types, but not for others. I also spotted memory corruptions ending up with wrong result (cached one). I am reverting that for now until this crate is stable.